### ISSUE TITLE
Design of error handling in akka-stream is inconsistent

### ISSUE THREAD

**Issue by cppexpert** [akka/akka#18359]
I'm building a library that is going to be used by 3rd-party. In one of my methods I return `Stream[Item]` that is asynchronously generated from result of paginated REST API call.

I'm using my modification of [BulkPullerAsync](https://github.com/MfgLabs/akka-stream-extensions/blob/master/commons/src/main/scala/internals/source/BulkPullerAsync.scala). My code is [here](https://gist.github.com/cppexpert/a366bf3fbd34b53b2e4c).

I want recipient of my stream to be able to handle errors. According to [documentation](http://doc.akka.io/docs/akka-stream-and-http-experimental/1.0/scala/stream-error.html) I'm supposed to use custom `Supervision.Decider`.

```scala
val decider: Supervision.Decider = {
  case ex =>
    ex.printStackTrace()
    Supervision.Stop
}

implicit val mat = ActorMaterializer(ActorMaterializerSettings(system).withSupervisionStrategy(decider))
```

Unfortunately it doesn't catch exceptions thrown in my `ActionPublisher`. I see it handled, `ActorPublisher.onError` is called but it doesn't reach `Supervision.Decider`. It works with simple `Stream` provided in documentation.

Errors also don't reach actor if I use `Sink.actorRef`.

What should I do ? I expect that users of my Stream should not depend on nature of its implementation.

And for the sake of experiment I tried following sample

```scala
val stream = Source(0 to 5).map(100 / _)
stream.runWith(Sink.actorSubscriber(props))
```

In this case exception was caught by `Decider` but never reached actor subscriber.

Overall I think it's inconsistent behavior. I cannot use one mechanism for handling errors in `Stream`.

PS. Original SO question: http://stackoverflow.com/q/32290285/226895

**Comment by cppexpert** [comment]
Seems like it's related to #16916

**Comment by rkuhn** [comment]
This will be solved by #19950. Operator supervision does not have sufficient coverage—as you note—and it cannot be extended to cover everything.
